desktop/core/src/desktop/auth/api_authentications.py

```diff
@@ -50,15 +50,26 @@ def authenticate(self, request):
     LOG.debug('JwtAuthentication: got access token %s' % access_token)
 
     try:
-      payload = jwt.decode(access_token, 'secret', algorithms=["HS256"])
+      payload = jwt.decode(
+        access_token,
+        'secret',
+        algorithms=["RS256"],
```

Comment:
Should we read alg from header?

>>> jwt.get_unverified_header('token')
{u'alg': u'RS256', u'typ': u'JWT', u'kid': u'1'}

Reply:
That is not the best practice @wing2fly as stated in the PyJWT docs since it will work for any custom JWT request coming to Hue (even attackers)!

Warning section in the `jwt.decode(...)` - https://pyjwt.readthedocs.io/en/stable/api.html#jwt.decode

But with this we can extract the `kid` when implementing the RSA public key fetching section since the kid will get verified anyway when getting the key from the server.